### DISCUSSION TITLE
Hai Reveal in the 0.10.2 Release is broken, correct?

### DISCUSSION THREAD

**Discussion by SomeTroglodyte** [endless-sky/endless-sky#9161] [Q&A] [answered]
No way to fulfill `has "Hai Leaks Response 2: done"` as there is no "Hai Leaks Response 2" anywhere, and no follow-up to `event "hr: exploration period"`, as there is no mention anywhere else.

Or did I overlook something?

**Comment by bene-dictator** [comment, marked as answer]
Yes, at the moment there's no mission named `Hai Leaks Response 2`, which upon completion would offer the mission `Hai Leaks Response 3`. In addition, the missions to continue the storyline (Yeertle, Ooonem) have been disabled.

I'd think (but I'm not certain) that if you want to continue the storyline, you'd want to change `Hai Leaks Response 3`'s to offer conditions, and remove all of the `disable mission` clause in `hai reveal 1 intro.txt`.

**Reply by SomeTroglodyte** [reply]
> remove all of the `disable mission` clause

Seems as long as I'm running the AppImage I can't do that - by save edit I could get "Hai Leaks Response 3" to run - but then the rest of the story would break again due to all those rescues being blocked (I count Unwanted Cargo as blocked)... Oh well, better forget about it then.

**Reply by SomeTroglodyte** [reply]
Note for casual readers - I did not forget about it after all, and it _can_ be resuscitated with a mod, while still running the AppImage - see below.

**Reply by oldpenguin5** [reply]
The attached workaround plugin continues the storyline and works for 0.10.6 and contiuous builds afterwards, also for an Appimage.
Create a directory .../endless-sky/plugins/fix-hai-reveal/data  and put the file there.

Missions Yeertle Family/Ooonem are disabled since 0.10.0.
In 0.10.1 missions Hai Leaks Response 1 & 2 were removed, with part of the content (human diplomats to Hai space) covered in file "hai reveal 0 prologue.txt" 

To continue the storyline after the prologue with Hai Leaks Response 3, "Hai Leaks Response 2: done" is a precondition.
The missions in this plugin allow to continue the storyline after the delayed end event "hr: exploration period" of the prologue.

[fix9161-hai-reveal.txt](https://github.com/endless-sky/endless-sky/files/15197057/fix9161-hai-reveal.txt)

Content-wise there is still a major discrepancy between the Hai reveal and the Wanderers storyline, because Sayari is newly introduced in both storylines even when already known from  the other.

**Reply by SomeTroglodyte** [reply]
All known, and I've been playing with a similar plugin for ages. Lost in Midnight also includes a deblocker IIRC, and mine and theirs don't even collide.
[repair-hai-reveal.zip](https://github.com/endless-sky/endless-sky/files/15197758/repair-hai-reveal.zip)
As you can see, timestamped September, and played through maybe 6 times with little snags since then - the snags being when the hints show.

As for the "puzzler" aspect - the recurring theme - that you so very often do not know whether you have finished or bungled a substory, or it is just waiting endlessly for a timer.. I got the idea that should be done declaratively. Missions get new optional "storyline" properties tying them together and controlling how status, hints, and so on are to be displayed. New tab on the logs panel contains only these storyline infos... I'd try it out, but code changes of that caliber need debugging capability and I got no idea how to best attach which debugger...

**Reply by Nova1422** [reply]
@SomeTroglodyte mind if I upload the plugin to the official plugin archive? All credits will be yours ofc, unless I manage to tweak something in testing it for the new versions

**Reply by SomeTroglodyte** [reply]
Mind? No, not really. It's possibly outdated, though:
- Haven't played in a long time
- I've lost my `.local/share/endless-sky` a while ago, so I don't know whether I fixed something after above post
- Not to mention adapting to ES version changes

And: I've stated in some places that content I show in an open source project may be reused by any contributor of said project freely and without attribution - as long as the result stays within that project.

**Reply by Nova1422** [reply]
I initially thought this would be a quick 10min fix and upload, maybe the fact that this plugin was years old, and as you mentioned later, incomplete as well, was overlooked by me 


Anyways, the incomplete parts were added, and your .txt file was updated and reduced (that is, some pre existing things were removed, as per I saw fit), the plugin will be uploaded in the official repo and linked here in a while

**Reply by Nova1422** [reply]
https://github.com/Nova1422/plugins?tab=readme-ov-file#unlockhaireveal it is, sending it for others dropping by here

**Comment by SomeTroglodyte** [comment]
In that case, it might have been better to remove the "hai reveal 0" file for release, that would have been a bit less confusing.

While the game is not in any way "done", this is the second-most irritating aspect of it: The UI and gameplay are smooth and polished, the FW story engrossing, so any perceived "dead ends" leave you puzzling: My fault? Did I miss something? Is it waiting on some impossible to guess trigger? Is it unfinished? Or is it _meant_ to be that way? Yottrite or the Sheragi station come to mind, while Gegno or Bunrodea are clear enough and didn't leave me scratching my head.

**Reply by bene-dictator** [reply]
The answer is that it was decided 0.10.0 should be ready for release before Hai Revealed was deemed to be ready, and all of the files for Hai Reveal were removed just for that release. It was originally a controversial decision given all of us working on it were not informed until it, well, happened – and because Hai Reveal at that point was largely functional and complete in that form even if it wasn't up to scratch.

After 0.10.0, Hai Reveal was to be released in small chunks while it was reworked and reworked again, and that is what we have now. The current introduction was reworked fairly quickly from higher up and there's work being done to put that rework back in line with what we had in mind – and that 0.10.2 was released in the middle of this rework-within-a-rework is more annoying.

There are still questions about the ending of Hai Reveal to be considered – the details are still to come. This process I am not really involved with as much like I had been last time.

Ssil Vida, Yottrite are simply dead-ends at the moment. Not your fault, I had the same feelings when I finished the current Remnant and Wanderer missions. There is still far more to come, though – perhaps we need to add a dialog window that shows up when you reach the end of a storyline.

**Comment by SomeTroglodyte** [comment]
I mean - the game could try to help. A little graph theory to automatically derive "story lines" (so no change to data required), then a question "yes my captain is not supposed to know this, but I agree to break the third wall (or was it fourth wall?)", and you get completion percentages and easier "log" access pertaining to a story so you could re-check what the ending text was... Oops, typed that before reading your post up to the "dialog" paragraph. Just rambling anyway. :+1:!

Still so much to discover - like how many RL years would it take blowing up Korath civilians to reach that "destroyer of worlds" rating, or can I take down that Heron 'cuz Zitchas keeps attacking me for no discernible reason??? :partying_face:

**Reply by lumbar527** [reply]
> Still so much to discover - like how many RL years would it take blowing up Korath civilians to reach that "destroyer of worlds" rating, or can I take down that Heron 'cuz Zitchas keeps attacking me for no discernible reason??? 🥳

Have you attacked any other author ships? Because Zitchas's person ship is of the standard author government.

**Comment by SomeTroglodyte** [comment]
"no discernible" means I'm not aware of attacking any non-hostiles, much less anything labeled "Author". But that's on another page - when does the "no friendly fire" assurance break expectations. I'm aware inadvertently leaving a "selection" on a friendly, flying away in the system, then firing a gun out of range and in the opposite direction, still counts as act of war. Also pretty sure I have nothing with an un-"safe" blast radius (No Sheragi Cannon). I can avoid those traps, no biggie, and not really immersion-breaking, as one can understand the reasons. Maybe they get hurt by the blast of one of my ships getting destroyed... Just another puzzle, not a negative for the overall quality.

**Comment by VaelophisNyx** [comment]
as a player, the fact that the HR mission overhaul was released in a state that leaves it impossible to continue without save/file editing, without telling you as much is fairly irritating

**Comment by SomeTroglodyte** [comment]
... Not that bad. It's mildly irritating that you can get so far in the Wanderer line without the Hai progressing, and that Sayari will be in two places at the same time (with or without HR). Or that you can get the Ooonem or Yeertle *Hints* but not the rescues themselves. And HR has severe errors like mistaking principles for principals (that's sarcasm).

That said, I simply made an unblock plugin, plays just fine - in 0.10.2 _or_ 0.10.3.
<details>

[repair-hai-reveal.zip](https://github.com/endless-sky/endless-sky/files/12656069/repair-hai-reveal.zip)
</details>

**Comment by RicCakes** [comment]
Just came back on the game again and I figured it'd be better to ask in a related thread.

Just got done with the FW storyline and rushed to Hai space to get some quick salary (Turner storyline) but I found out it's behind Hai Reveal now (tho imo I think it still makes sense even without the Hai leaks). That being said, does this sudden new storyline well integrated with the other storylines? Like the Wanderers storyline I mean

**Reply by bene-dictator** [reply]
I've been thinking about decoupling the Turner missions again.

**Comment by SomeTroglodyte** [comment]
> sudden new storyline

What? Where?

**Reply by RicCakes** [reply]
My bad, I just came back after a year and there's a lot of new stuff (really nice seeing the game going forward), the new storyline I'm mentioning is the whole Hai Reveal storyline (I see all the chapters in the data file but a post here says its only up to prologue at the moment). and I was wondering if it works well with the Wanderer Storyline since I saw it has Sayari in the missions when I took a peek.

**Reply by SomeTroglodyte** [reply]
OK, as I said I play with above mod that just "unplugs" the stoppers, and the contradictions are there, but no worse than pure release.

**Reply by RicCakes** [reply]
Alright, I'll check it out

**Reply by SomeTroglodyte** [reply]
It might feel strange where it just skips what might be missing between "hr: secret leaks" and "Hai Leaks Response 2" - but I was too lazy to dig through history here to find a missing "Hai Leaks Response 1" text. Might do later. I also made some things easier to trigger - see comments in the file: like Unwanted Cargo and Nanachi. Missions with <1% probability are fine if nothing depends on them, from my experience with the Alitis one I estimate you should get those just fine without metagaming every 5 to 10 playthroughs. But HR suffers seriously when you miss those. So I changed them too.

Lastly - not redistribution ready as I was too lazy yet to clean up license and attributions. Just finished the test run. Almost. There may be more Turner...
